### PR TITLE
[UCAI] Add callable support for python function creation

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

This PR introduces the ability to provide a Google PyDoc formatted python function with type hints to the client's `create_python_function` API to generate a UC function from a base callable. 

Note: generic types, Union types, class methods, instance methods, and generic collection types are not permitted for registry due to the need for strong typing within UC.

### CODE REVIEW

**Comment by BenWilson2** [comment]
@serena-ruan @dennyglee This one's ready!

File: unitycatalog-ai/src/ucai/core/client.py [hunk 1]
```diff
@@ -42,6 +45,23 @@ def __init__(self, client: Any = None, **kwargs: Any) -> None:  # noqa: ARG002
     @abstractmethod
     def create_function(self, *args: Any, **kwargs: Any) -> Any:
         """Create a function"""
+    
+    @abstractmethod
+    def create_python_function(
+        self, *, func: Callable[..., Any], catalog: str, schema: str, replace: bool = False
+    ) -> "FunctionInfo":
```
Comment (serena-ruan) [inline unitycatalog-ai/src/ucai/core/client.py:52]: Can we avoid importing FunctionInfo here? As this is used by OSS client as well, we don't want to introduce databricks dependency in general :)

**Review by serena-ruan — APPROVED** [review]
Everything else LGTM! Let's add Databricks client support for this after https://github.com/unitycatalog/unitycatalog/pull/578 is merged :)

File: unitycatalog-ai/src/ucai/core/client.py (hunk 1 above)
Comment (BenWilson2) [reply unitycatalog-ai/src/ucai/core/client.py:52]: Yep! Sounds good!

**Comment by BenWilson2** [comment]
@dennyglee we're all good to merge this one

**Review by dennyglee — APPROVED** [review]
👍

**Pull request merged**